functions/prepared-check-sync/check.data.service.js

```diff
@@ -0,0 +1,57 @@
+const R = require('ramda')
+const sqlService = require('../lib/sql/sql.service')
+const { TYPES } = sqlService
+
+/**
+ * Find latest live and single try out checkCode for a particular pupil based on one of the pupil's checkCodes
+ * @param checkCode
+ * @return {Promise<Array>}
+ */
+module.exports.sqlFindActiveCheckCodesByCheckCode = async function (checkCode) {
+  const sql = `          
+  DECLARE
+      @pupilId INT = (SELECT pupil_id
+                      FROM [mtc_admin].[check] chk
+                      WHERE chk.checkCode = @checkCode)
```

Comment:
What happens if this returns null?

Reply:
the query will return an object with null values and do nothing since I am filtering these.